src/views/docs-view/utils/get-root-docs-path-generation-functions.ts

```diff
@@ -0,0 +1,54 @@
+import { getStaticGenerationFunctions } from 'views/docs-view/server'
+import { ProductSlug } from 'types/products'
+import { GetStaticPaths, GetStaticProps } from 'next'
+import { cachedGetProductData } from 'lib/get-product-data'
+import { removeIndexPath } from 'lib/remove-index-path'
+
+/**
+ * Generates static functions for use in a
+ * `pages/<productSlug>/<rootDocsPath>/[...page].tsx` DocsView page file.
+ */
+export function getRootDocsPathGenerationFunctions(
```

Comment:
This is a light wrapper around `views/docs-view/server`. Aiming to consolidate the two in #750, but wanted to keep this PR scoped to `terraform` only.